test/libsolidity/syntaxTests/dataLocations/libraries/library_function_with_data_location_transient.sol

```diff
@@ -0,0 +1,19 @@
+library L {
+    function f(uint[] transient) private pure returns (uint[] transient) { }
+    function g(uint[] transient) internal pure returns (uint[] transient) { }
+    function h(uint[] transient) public pure returns (uint[] transient) { }
+    function i(uint[] transient) external pure returns (uint[] transient) { }
+}
+// ----
+// DeclarationError 2333: (67-83): Identifier already declared.
+// DeclarationError 2333: (145-161): Identifier already declared.
+// DeclarationError 2333: (221-237): Identifier already declared.
+// DeclarationError 2333: (299-315): Identifier already declared.
```

Comment:
It would be best if the only invalid thing in the test was the thing you're testing for, i.e. the `transient` location. To avoid the `DeclarationError` here you could test parameters and return values separately (i.e. have a separate function for each).

```suggestion
library L {
    function fin(uint[] transient) private pure {}
    function gin(uint[] transient) internal pure {}
    function hin(uint[] transient) public pure {}
    function iin(uint[] transient) external pure {}

    function fout() private pure returns (uint[] transient) {}
    function gout() internal pure returns (uint[] transient) {}
    function hout() public pure returns (uint[] transient) {}
    function iout() external pure returns (uint[] transient) {}
}
// ----
```

Reply:
Done.